assets/js/googlesitekit-adminbar.js

```diff
@@ -26,109 +26,92 @@ import './modules';
 /**
  * WordPress dependencies
  */
-import { Component, render, Fragment } from '@wordpress/element';
+import { render, Fragment, useCallback } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies.
  */
+import Data from 'googlesitekit-data';
 import {
 	decodeHtmlEntity,
-	getSiteKitAdminURL,
 	loadTranslations,
 	trackEvent,
 } from './util';
 import Link from './components/link';
 import AdminbarModules from './components/adminbar/adminbar-modules';
 import Root from './components/root';
+import { STORE_NAME as CORE_SITE } from './googlesitekit/datastore/site/constants';
+const { useSelect } = Data;
 
-export class GoogleSitekitAdminbar extends Component {
-	constructor( props ) {
-		super( props );
-
-		this.handleMoreDetailsLink = this.handleMoreDetailsLink.bind( this );
-	}
-
-	async handleMoreDetailsLink() {
-		const {
-			permaLink,
-			postID,
-			postType,
-			pageTitle,
-		} = global._googlesitekitLegacyData;
-		const href = getSiteKitAdminURL(
-			'googlesitekit-dashboard',
-			{
-				permaLink,
-				id: postID,
-				postType,
-				pageTitle,
-			}
-		);
+export const GoogleSitekitAdminbar = () => {
+	const dashboardURL = useSelect( ( select ) => select( CORE_SITE ).getAdminURL( 'googlesitekit-dashboard' ) );
+	const currentEntityURL = useSelect( ( select ) => select( CORE_SITE ).getCurrentEntityURL() );
+	const currentEntityTitle = useSelect( ( select ) => select( CORE_SITE ).getCurrentEntityTitle() );
 
+	const onMoreDetailsClick = useCallback( async () => {
+		const detailsURL = addQueryArgs( dashboardURL, { permaLink: currentEntityURL } );
 		await trackEvent( 'admin_bar', 'post_details_click' );
-		document.location = href;
-	}
+		document.location = detailsURL;
```

Comment:
Since we're refactoring this, let's use `assign` here as is our convention.
```suggestion
		document.location.assign( detailsURL );
```